sql/parser: remove anon ArgTypes

dt opened the pull request:
This switches the remaining uses of `ArgTypes` to `NamedArgTypes` and removes the un-named.
This is an alternative approach to removing the duplication between the two addressed in #13032.

Closes #13032.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13046)
<!-- Reviewable:end -->

dt committed:
sql/parser: remove anon ArgTypes

this switches the remaining uses to NamedArgTypes and removes the last few remaining uses of the un-named version.

Closes #13032.

nvanbenschoten commented:
Even better! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

Do you think this should be followed up with a `gorename -from '"github.com/cockroachdb/cockroach/pkg/sql/parser".NamedArgTypes' -to ArgTypes`?

---

Review status: 0 of 8 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13046#-:-Kb9NljzSwLBMMERflSr:b-ff0pgf)*
<!-- Sent from Reviewable.io -->

knz reviewed (approved):
That's a nice feature :+1:

dt committed:
sql/parser: rename NamedArgTypes to ArgTypes

Now that the named version is the only on, the specifier isn't needed.

Purely mechanical change via:
	`gorename -from 'github.com/cockroachdb/cockroach/pkg/sql/parser.NamedArgTypes' -to ArgTypes`

dt commented:
Good idea, added the rename.
